src/Oracle/Protocol/OracleHttpProtocol.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JArray = Newtonsoft.Json.Linq.JArray;
+using JObject = Newtonsoft.Json.Linq.JObject;
+
+namespace Neo.Plugins
+{
+    public class OracleHttpProtocol : IOracleProtocol
+    {
+        public int Timeout { get; set; } = 5000;
+        public bool AllowPrivateHost { get; set; } = false;
+        public readonly string[] AllowedFormats = new string[] { "application/json" };
+
+        public byte[] Request(ulong requestId, string url, string filter)
+        {
+            Utility.Log(nameof(OracleHttpProtocol), LogLevel.Debug, $"Request: {url}");
+
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri)) throw new InvalidOperationException("UrlError");
+            if (!AllowPrivateHost && IsInternal(Dns.GetHostEntry(uri.Host))) throw new InvalidOperationException("Access to private host is not allowed");
+
+            using var handler = new HttpClientHandler
+            {
+                // TODO: Accept all certificates
+                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
```

Comment:
We don't want to merge it with this, do we?

Reply:
It should be used only for UT

Reply:
@superboyiii we should change this

Reply:
> @superboyiii we should change this

@Tommo-L

Reply:
Done